dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -1602,20 +1605,26 @@ private Response fireAction(final HttpServletRequest request,
     private void processPermissions(final FireActionForm fireActionForm,
                                     final ContentletDependencies.Builder formBuilder) {
 
-        if (UtilMethods.isSet(fireActionForm.getIndividualPermissions())) {
+        if (null != fireActionForm.getIndividualPermissions()) {
 
             final List<Permission> permissions = new ArrayList<>();
             for(final Map.Entry<PermissionAPI.Type, List<String>> entry :
                     fireActionForm.getIndividualPermissions().entrySet()) {
 
                 entry.getValue().forEach(roleId -> permissions.add(
-                        new Permission(null, roleId, entry.getKey().getType())));
+                        new Permission(null, this.mapRoleId(roleId), entry.getKey().getType())));
             }
 
             formBuilder.permissions(permissions);
         }
     }
 
+    private String mapRoleId (final String roleIdOrKey) {
```

Comment:
we would need a postman or IT to validate that:
- we can support roleId/roleKey
- if the roleId/roleKey is invalid, we ignore it and continue with the remaining permissions

Reply:
Done